platform/ext/target/nordic_nrf/common/native_drivers/spu.c

```diff
@@ -0,0 +1,130 @@
+	(FLASH_TOTAL_SIZE / FLASH_SECURE_ATTRIBUTION_REGION_SIZE)
+#define NUM_SRAM_SECURE_ATTRIBUTION_REGIONS \
+	(TOTAL_RAM_SIZE / SRAM_SECURE_ATTRIBUTION_REGION_SIZE)
+
+/* Convenience macros for SPU NCS attribution */
+#define FLASH_NSC_REGION_FROM_ADDR(addr) \
+	((u32_t)addr / FLASH_SECURE_ATTRIBUTION_REGION_SIZE)
+
+#define FLASH_NSC_SIZE_FROM_ADDR(addr) (FLASH_SECURE_ATTRIBUTION_REGION_SIZE \
+	- (((u32_t)(addr)) % FLASH_SECURE_ATTRIBUTION_REGION_SIZE))
+
+#define FLASH_NSC_SIZE_REG(size) ((31 - __builtin_clz(size)) - 4)
+
+
+void spu_enable_interrupts(void)
+{
+	nrf_spu_int_enable(NRF_SPU,
+		NRF_SPU_INT_FLASHACCERR_MASK |
+		NRF_SPU_INT_RAMACCERR_MASK |
+		NRF_SPU_INT_PERIPHACCERR_MASK);
+}
+
+void spu_regions_reset_all_secure(void)
+{
+	for (size_t i = 0; i < NUM_FLASH_SECURE_ATTRIBUTION_REGIONS ; i++) {
+		nrf_spu_flashregion_set(NRF_SPU, i,
+			1 /* Secure */,
+			NRF_SPU_MEM_PERM_READ
+			| NRF_SPU_MEM_PERM_WRITE
+			| NRF_SPU_MEM_PERM_EXECUTE,
+			0 /* No lock */);
+	}
+
+	for (size_t i = 0; i < NUM_SRAM_SECURE_ATTRIBUTION_REGIONS ; i++) {
+		nrf_spu_ramregion_set(NRF_SPU, i,
+			1 /* Secure */,
+			NRF_SPU_MEM_PERM_READ
+			| NRF_SPU_MEM_PERM_WRITE
+			| NRF_SPU_MEM_PERM_EXECUTE,
+			0 /* No lock */);
+	}
+}
+
+void spu_regions_config_non_secure(
+    u32_t start_addr,
+    u32_t limit_addr,
+    bool ram)
+{
+	if (ram) {
+		/* Determine start and last ram region number */
+		size_t start_id =
+			start_addr / SRAM_SECURE_ATTRIBUTION_REGION_SIZE;
+		size_t last_id =
+			limit_addr / SRAM_SECURE_ATTRIBUTION_REGION_SIZE;
+
+		/* Configure all regions between start_id and last_id */
+		for (size_t i = start_id; i <= last_id; i++) {
+			nrf_spu_ramregion_set(NRF_SPU, i,
+				0 /* Non-Secure */,
+				NRF_SPU_MEM_PERM_READ
+				| NRF_SPU_MEM_PERM_WRITE
+				| NRF_SPU_MEM_PERM_EXECUTE,
+				1 /* Lock */);
+		}
+	} else {
+		/* Determine start and last flash region number */
+		size_t start_id =
+			start_addr / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+		size_t last_id =
+			limit_addr / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+
+		/* Configure all regions between start_id and last_id */
+		for (size_t i = start_id; i <= last_id; i++) {
+			nrf_spu_flashregion_set(NRF_SPU, i,
+				0 /* Non-Secure */,
+				NRF_SPU_MEM_PERM_READ
+				| NRF_SPU_MEM_PERM_WRITE
+				| NRF_SPU_MEM_PERM_EXECUTE,
+				1 /* Lock */);
+		}
+	}
+}
+
+void spu_regions_config_non_secure_callable_flash(u32_t start_addr, u32_t limit_addr)
+{
+	size_t size = limit_addr - start_addr + 1;
+
+	u32_t nsc_size = FLASH_NSC_SIZE_FROM_ADDR(start_addr);
+
+	/* Check Non-Secure Callable region possible overflow */
+	NRFX_ASSERT(size <= nsc_size);
+
+	/* Check Non-Secure Callable region ending on SPU boundary */
+	NRFX_ASSERT(((start_addr + size) %
+			FLASH_SECURE_ATTRIBUTION_REGION_SIZE) == 0);
+
+	/* Check Non-Secure Callable region power-of-2 size compliance */
+	NRFX_ASSERT((nsc_size & (nsc_size - 1)) == 0,
+		"The Non-Secure Callable region size not power-of-2\n");
```

Comment:
`NRFX_ASSERT` takes only one parameter.

Reply:
Yeap, that was forgotten; corrected it now